File: javascript/6_kyu/valid_phone_number.js
// Valid Phone Number
// https://www.codewars.com/kata/525f47c79f2f25a4db000025/train/javascript

function validPhoneNumber(s) {
    s = s.split("");
    var result = [];
    var template = "(xxx) xxx-xxxx";
    var digits = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "0"]

    for (var i = 0; i < s.length; i++) {
        if ( digits.includes(s[i]) ) {
            result.push("x");
        } else {
            result.push(s[i]);
        }
    }

    return result.join("") == template;
}

console.log(validPhoneNumber("(123) 456-7890"));
console.log(validPhoneNumber("(1111)555 2345"));
console.log(validPhoneNumber("(098) 123 4567"));

File: python/6_kyu/valid_phone_number.py
# Valid Phone Number
# https://www.codewars.com/kata/525f47c79f2f25a4db000025/train/python

def validPhoneNumber(s):
    s = list(s)
    if len(s) == 14:
        if s[0] == "(" and s[4] == ")" and s[5] == " " and s[9] == "-":
            digit_index = [1, 2, 3, 6, 7, 8, 10, 11, 12, 13]
            for i in range(len(s)):
                if i in digit_index:
                    if s[i].isdigit():
                        return True
                    else:
                        return False
        else:
            return False
    else:
        return False

print(validPhoneNumber("(123) 456-7890"))
print(validPhoneNumber("(1111)555 2345"))
print(validPhoneNumber("(098) 123 4567"))